### ISSUE TITLE
test-output-multi-tags failing because of comments in JSON

### ISSUE THREAD

**Issue by alex85k** [openstreetmap/osm2pgsql#481]
For a long time I had failing test on Windows (and, if I remember correctly, on other OSes too):

```
`test 14
    Start 14: test-output-multi-tags
14: Test command: D:\winbuilds_d\osm2pgsql\build\tests\Debug\test-output-multi-tags.exe
14: Test timeout computed to be: 1200
14: WARNING: osm2pgsql was compiled without fork, only using one process!
14: ERROR: Unable to parse multi config file `tests/test_output_multi_tags.json': tests/test_output_multi_tags.json(1): expected value
```

After debugging I founded that parser means exactly what it prints: it can not read line 1 of this file:
https://github.com/openstreetmap/osm2pgsql/blob/master/tests/test_output_multi_tags.json
because it is a comment and it seems that boost does not support it. After removing the comments, the tests pass.

**Comment by alex85k** [comment]
I have created the issue instead of PR because that comments are useful for understanding the process.

**Comment by alex85k** [comment]
Just checked - there is no option to skip Json comments in Boost PropertyTree. Moving commented json into docs/multi.md seems to be the best solution (better done by author).

**Comment by pnorman** [comment]
JSON doesn't have comments, which is one reason why I'm not writing my complex multi-backend styles directly in JSON. 

Oddly, [multi.style.json](https://github.com/openstreetmap/osm2pgsql/blob/master/multi.style.json) **does** work for me, with a comment at the start. It also states

> It contains comments so is invalid JSON but the parser doesn't complain.

So I guess some versions of boost complain on some operating systems. 

There are inline comments too, those are going to be hard to move to docs/multi.md

**Comment by alex85k** [comment]
It was broken with Boost 1.59 only:
https://github.com/boostorg/property_tree/commit/b2c488be9bff51eb903136dae08235ee8e36fcf9

**Comment by lonvia** [comment]
My preference would be to move comments into the corresponding .cc test files.

**Comment by pnorman** [comment]
Moving the comments works for the test files, but not multi.style.json. We could move the header into multi.lua, but there's stuff like

```js
      /* By using boolean columns stylesheet processing can be simplified */
      {"name": "shelter", "type": "boolean"},
      {"name": "bench", "type": "boolean"},
      {"name": "wheelchair", "type": "boolean"}
```

The comment is explaining a feature, and detaching it makes it much less useful.

I suppose we could move to XML or INI, the other two formats boost::propertytree has support for.

**Comment by alex85k** [comment]
You can just use  fake field ```"comment" : "By using boolean columns stylesheet ..." and ignore it while analyzing. http://stackoverflow.com/questions/244777/can-i-use-comments-inside-a-json-file
 :

**Comment by lonvia** [comment]
Fixed in bc411ffd54231c5f3e103f1fb5db483c46089665.
